runtime/integration-tests/src/generic/cases/proxy.rs

```diff
@@ -0,0 +1,236 @@
+use cfg_primitives::Balance;
+use frame_support::traits::Get;
+use frame_system::RawOrigin;
+use sp_runtime::{traits::StaticLookup, DispatchResult};
+use xcm::v3::{MultiLocation, WeightLimit};
+
+use crate::{
+	generic::{
+		config::Runtime,
+		env::{Blocks, Env},
+		envs::{fudge_env::FudgeSupport, runtime_env::RuntimeEnv},
+		utils::{
+			self,
+			currency::{self, cfg, usd6, CurrencyInfo, Usd6},
+			genesis::{self, Genesis},
+			POOL_MIN_EPOCH_TIME,
+		},
+	},
+	utils::{accounts::Keyring, tokens::rate_from_percent},
+};
+
+const FROM: Keyring = Keyring::Charlie;
+const PROXY: Keyring = Keyring::Alice;
+const TO: Keyring = Keyring::Bob;
+
+const FOR_FEES: Balance = cfg(1);
+const TRANSFER_AMOUNT: Balance = usd6(100);
+
+fn configure_proxy_and_transfer<T: Runtime>(proxy_type: T::ProxyType) -> DispatchResult {
+	let mut env = RuntimeEnv::<T>::from_parachain_storage(
+		Genesis::<T>::default()
+			.add(genesis::balances(T::ExistentialDeposit::get() + FOR_FEES))
+			.add(genesis::tokens(vec![(Usd6::ID, Usd6::ED)]))
+			.storage(),
+	);
+
+	let call = pallet_restricted_tokens::Call::transfer {
+		currency_id: Usd6::ID,
+		amount: TRANSFER_AMOUNT,
+		dest: T::Lookup::unlookup(TO.id()),
+	}
+	.into();
+
+	configure_proxy_and_call::<T>(env, proxy_type, call)
+}
+
+fn configure_proxy_and_x_transfer<T: Runtime>(proxy_type: T::ProxyType) -> DispatchResult {
+	/*
+	let env = // TODO: initialize fudge env with xcm support
+	let call = // TODO: create a 'pallet_restricted_xtokens::transfer' call
+
+	configure_proxy_and_call::<T>(env, proxy_type, call);
+	*/
+	todo!()
+}
```

Comment:
@cdamian I have no knowledge of the setup for a xcm transfer, could you finish this test for `pallet_restricted_xtokens`? It should be exactly the same pattern as the one already done for `pallet_restricted_tokens` but using `FudgeEnv` instead. 🙏🏻